packages/nimble-components/src/table-column/enum-text/index.ts

```diff
@@ -0,0 +1,128 @@
+import { DesignSystem } from '@microsoft/fast-foundation';
+import { attr } from '@microsoft/fast-element';
+import {
+    MappingConfig,
+    TableColumnEnumBase,
+    TableColumnEnumColumnConfig
+} from '../enum-base';
+import { styles } from '../enum-base/styles';
+import { template } from '../enum-base/template';
+import { TableColumnSortOperation, TableColumnValidity } from '../base/types';
+import { mixinGroupableColumnAPI } from '../mixins/groupable-column';
+import { mixinFractionalWidthColumnAPI } from '../mixins/fractional-width-column';
+import { Mapping } from '../../mapping/base';
+import { MappingText } from '../../mapping/text';
+import {
+    enumTextColumnValidityFlagNames,
+    TableColumnEnumTextValidator
+} from './models/column-validator';
+import type { ColumnInternalsOptions } from '../base/models/column-internals';
+import { tableColumnEnumTextCellViewTag } from './cell-view';
+import { tableColumnEnumTextGroupHeaderViewTag } from './group-header-view';
+
+declare global {
+    interface HTMLElementTagNameMap {
+        'nimble-table-column-enum-text': TableColumnEnumText;
+    }
+}
+
+export interface TableColumnEnumTextColumnConfig
+    extends TableColumnEnumColumnConfig {
+    placeholder: string;
+}
+
+/**
+ * Table column that maps values to strings
+ */
+export class TableColumnEnumText extends mixinGroupableColumnAPI(
+    mixinFractionalWidthColumnAPI(
+        TableColumnEnumBase<TableColumnEnumTextColumnConfig>
+    )
+) {
+    @attr
+    public placeholder?: string;
+
+    protected get supportedMappingTypes(): readonly (typeof Mapping)[] {
+        return [MappingText] as const;
+    }
+
+    private readonly validator: TableColumnEnumTextValidator = new TableColumnEnumTextValidator(
+        this.columnInternals,
+        enumTextColumnValidityFlagNames
+    );
+
+    public override get validity(): TableColumnValidity {
+        return this.validator.getValidity();
+    }
+
+    /**
+     * @internal
+     */
+    public override handleChange(source: unknown, args: unknown): void {
+        super.handleChange(source, args);
+        if (source instanceof Mapping && typeof args === 'string') {
+            if (args === 'key') {
+                this.validateKeyDependentConditions();
+            }
+        }
+    }
+
+    protected override getColumnInternalsOptions(): ColumnInternalsOptions {
+        return {
+            cellRecordFieldNames: ['value'],
+            cellViewTag: tableColumnEnumTextCellViewTag,
+            groupHeaderViewTag: tableColumnEnumTextGroupHeaderViewTag,
+            delegatedEvents: [],
+            sortOperation: TableColumnSortOperation.basic
+        };
+    }
+
+    protected override mappingsChanged(): void {
+        super.mappingsChanged();
+        this.validator?.validateAtMostOneDefaultMapping(this.mappings);
```

Comment:
The issue is that the `mappings` or `keyType` can get initialized before `validator` is initialized. When that happens, some of these functions can be called while `validator` is null, even though it is not typed as allowing null.

Reply:
A few comments/concerns about this approach:
- I'm surprised this isn't causing linting errors due to an unnecessary condition.
- I wouldn't be surprised if it causes problems because the TypeScript compiler might take out that check because it thinks it is unnecessary.
- If `validator` becomes defined after `mappings` and `keyType` are initialized, is there anything that then causes validation to run after it had been skipped?

Is the timing related to being connected? Could you skip validation when the column isn't connected and then perform validation in the connected callback?

Reply:
If I remember right, it was trying to run validation on the initializer values for `mappings` (`[]`) and `keyType` (`'string'`), so it would still get run again for the values coming from the html. However, it seems safer to do what you suggest and just do an initial validation at connection time.